akka-distributed-data/src/main/scala/akka/cluster/ddata/ReplicatedData.scala

```diff
@@ -37,6 +37,17 @@ trait ReplicatedData {
 
 }
 
+trait DeltaReplicatedData extends ReplicatedData {
+  type D <: ReplicatedData
```

Comment:
I can't think of an example like this. It's also easier to think about the deltas always moving the type between "complete" states than some partial one. But I've not read that many CRDT papers, just a few so maybe I'm missing an use case.